### ISSUE TITLE
plonk.visualize.plot passes colorbar_kwargs to imshow

### ISSUE THREAD

**Issue by joshcalcino** [dmentipl/plonk#21]
I am trying to change the position of the colorbar to the left hand side of the plot. I think I should be able to do this by setting `colorbar_kwargs={'position': 'left'}`, but this raises the error
```AttributeError: 'AxesImage' object has no property 'colorbar_kawrgs'``` 

It seems that `colorbar_kwargs` is getting passed to the `plots.imshow` function, which raises this error.

The code below reproduces the error
```
dumpfile = 'internal/e40_q2_gtd/rescale_hires_00092.h5'

snap= plonk.load_snap(dumpfile)

RENDER = 'density'

dens = plonk.visualize.plot(snap=snap, quantity=RENDER,
                     show_colorbar=True, cmap='gist_heat', norm='log', colorbar_kwargs={'position': 'left'})
```

Also another thing I noticed in the documentation for `plonk.visualize.plot`, it states it returns a matplotlib axes object:

```
    Returns
    -------
    ax
        The matplotlib Axes object.
```
However, this does not seem to be the case:

```
>>> print(dens)
<plonk.Visualization "rescale_hires_00092.h5">
```

I am also wondering if it is possible to make the `plot_object` object available to the user somehow in the above object, or make it accessible some other way. It would be useful if one wants to have more control over other matplotlib features themselves. For example, if I wanted to customise the colorbar after making the plot, I need `plot_object` to do so.

**Comment by dmentipl** [comment]
Hi @joshcalcino, thanks for raising this issue.

Which version of Plonk are you using? You can get this by printing `plonk.__version__`.

I have released a new version of Plonk (v0.5.0) today. In this version I have removed the `Visualization` class in favour of a function (`plonk.visualize.plot`) that returns a Matplotlib Axes object. I think this makes the code simpler, and more general. And it's easier for the user who might be already familiar with Matplotlib.

I've tested your code with v0.5.0 and it works as expected.

> Also another thing I noticed in the documentation for plonk.visualize.plot, it states it returns a matplotlib axes object:

On readthedocs there are two versions: one that is pinned to the most recent release installable via pip or conda (<https://plonk.readthedocs.io/en/stable/>) which is called "stable" and one that is the current master branch of the development version on GitHub (<https://plonk.readthedocs.io/en/latest/>) that is called "latest". I think you were looking at the development version, i.e. "latest", not the "stable" one.

> I am also wondering if it is possible to make the plot_object object available to the user somehow in the above object, or make it accessible some other way.

I guess now that there is no `Visualization` object you can just access the required Matplotlib objects via the Axes object. (Although you could do this previously.) For example, to change the colorbar range:

```python
ax = plonk.visualize.plot(snap=snap, quantity='density')
im = ax.images[0]
im.set_clim(vmax=1/137)
```

Let me know what you think.

**Comment by dmentipl** [comment]
Hi @joshcalcino , let me know if you have any more questions or concerns regarding this issue. If not, I'll close the issue.
